ext/spl/spl_directory.c

```diff
@@ -2740,8 +2746,12 @@ PHP_METHOD(SplFileObject, __toString)
 
 	CHECK_SPL_FILE_OBJECT_IS_INITIALIZED(intern);
 
-	if (!intern->u.file.current_line && Z_ISUNDEF(intern->u.file.current_zval)) {
-		spl_filesystem_file_read_line(ZEND_THIS, intern);
+	if (!intern->u.file.current_line) {
+		ZEND_ASSERT(Z_ISUNDEF(intern->u.file.current_zval));
```

Comment:
One thing that confused me is the original check: `!intern->u.file.current_line && Z_ISUNDEF(intern->u.file.current_zval)`. I think that if one of these conditions in the `&&` is false, the other one is as well.
If that is not the case then it will still be buggy (in other places too) because there could be a potential NULL deref if it is possible that `Z_ISUNDEF(intern->u.file.current_zval)` is false while `!intern->u.file.current_line` is true.
That's why I changed it to an assert.

Reply:
Isn't it possible if one does some rewind or something like that? But yeah that does seem lightly weird otherwise

Reply:
Hm I don't think so, rewind calls `spl_filesystem_file_free_line` which (confusingly) also clears out `current_zval`...

Reply:
Yeah than this should be fine